04_numeronym/alextmz/main_test.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_numeronyms(t *testing.T) {
+	var tests = map[string]struct {
+		arg  []string
+		want []string
+	}{
+		"empty string": {
```

Comment:
You should write two test cases, one for `TestNumeronyms` and one for `TestNumeronym`. The `TestNumeronyms` test case should not have this much detail, because all the `numeronyms` function does is call `numeronym` on a slice of strings and returns the result as a slice. That is all you should be testing what works. In this case, I'd be testing the empty slice, and a slice with "some number" of elements. That would be enough to satisfy you that `numeronyms` is doing what it claims to do.

In `TestNumeronym`, you should be testing the function that converts one string into another. You would use a set of test cases to exercise the function well, but you wouldn't have all these slices of strings as you do in this test case. That would make it easier to read. Just pairs of inputs and expected values.

In the case that a test fails, this will make it easier to isolate the failure.